frontend/components/puzzle4.jsx

```diff
@@ -34,9 +35,23 @@ class Puzzle4 extends React.Component {
     });
   }
 
+  submitAnswer() {
```

Comment:
we need to reset when the answer is submitted wrong, or is that coming in a separate PR?

Reply:
oh that's true, I'll reset state in the submitAnswer function too

Reply:
I currently have two instances where I call this.clearValue(). Left it there currently pending on the potential refactor using getCurrentStage. Still, don't understand what the refactor looks like